### PR TITLE
Update to bitcoin 0.16.1 release (constant scriptCode policy in non-segwit scripts).

### PR DESCRIPTION
**Pull request by alexdupre**

### CODE REVIEW

File: src/main/scala/fr/acinq/bitcoin/Script.scala [hunk 1]
```diff
@@ -294,12 +299,16 @@ object Script {
 
   def isPayToScript(script: Array[Byte]): Boolean = script.length == 23 && script(0) == elt2code(OP_HASH160).toByte && script(1) == 0x14 && script(22) == elt2code(OP_EQUAL).toByte
 
-  def removeSignature(script: List[ScriptElt], signature: BinaryData): List[ScriptElt] = {
+  def removeSignature(script: List[ScriptElt], signature: BinaryData): (List[ScriptElt], Boolean) = {
```
Comment (sstone) [inline src/main/scala/fr/acinq/bitcoin/Script.scala]: Is it really necessary ? You can tell is something has been removed by comparing the output and input

File: src/main/scala/fr/acinq/bitcoin/Script.scala [hunk 2]
```diff
@@ -294,12 +299,16 @@ object Script {
...
-    script.filterNot(_ == toRemove)
+    (script.filterNot(_ == toRemove), script.contains(toRemove))
   }
 
-  def removeSignatures(script: List[ScriptElt], sigs: List[BinaryData]): List[ScriptElt] = sigs.foldLeft(script)(removeSignature)
+  def removeSignatures(script: List[ScriptElt], sigs: List[BinaryData]): (List[ScriptElt], Int) = sigs.foldLeft((script, 0)) {
+    case (acc, sig) =>
+      val (updated, found) = removeSignature(acc._1, sig)
+      (updated, if (found) acc._2 + 1 else acc._2)
+  }
```
Comment (sstone) [inline src/main/scala/fr/acinq/bitcoin/Script.scala]: Same comment as above, plus if feels strange to use a counter here and a boolean in removeSignature (why not also a counter there ?)

**Comment by sstone** [comment]
Which test vectors are you using ? It seems that they're not from 0.16.1 but from a later commit ?

**Comment by alexdupre** [comment]
They are taken from the v0.16.1 release, why do you say they aren't?

**Comment by sstone** [comment]
`script_tests.json` seems to be different, it would be more consistent to use 0.16.1's version

**Comment by alexdupre** [comment]
Fully synced now

**Pull request merged**